File: sdk/py/hop_top_kit/tui/confirm.py
"""hop_top_kit.tui.confirm — yes/no confirmation prompt."""

from __future__ import annotations

import questionary
from questionary import Style

from hop_top_kit.cli import Theme


def confirm(theme: Theme, message: str, default: bool = True) -> bool:
    """Prompt user for yes/no confirmation.

    Args:
        theme:   Theme supplying accent color for the pointer.
        message: Prompt text shown to the user.
        default: Pre-selected answer (True = yes, False = no).

    Returns:
        True if confirmed, False otherwise.

    Raises:
        KeyboardInterrupt: on Ctrl-C or if questionary returns None.
    """
    style = Style(
        [
            ("qmark", f"fg:{theme.accent} bold"),
            ("question", "bold"),
            ("answer", f"fg:{theme.accent} bold"),
            ("pointer", f"fg:{theme.accent} bold"),
            ("highlighted", f"fg:{theme.accent} bold"),
            ("selected", f"fg:{theme.accent}"),
        ]
    )

    result = questionary.confirm(message, default=default, style=style).ask()

    if result is None:
        raise KeyboardInterrupt

    return result

File: sdk/ts/src/tui/confirm.ts
/**
 * @module tui/confirm
 * @package @hop-top/kit
 *
 * Interactive yes/no confirmation prompt.
 * Wraps @clack/prompts confirm(); throws on cancel (Ctrl-C).
 * Mirrors Go's NewConfirm from hop.top/kit/tui.
 */

import { confirm as clackConfirm, isCancel } from '@clack/prompts';
import type { Theme } from '../cli.js';

export interface ConfirmOpts {
  /** Default selection when user presses Enter without input. */
  initial?: boolean;
}

/**
 * Prompt the user with a yes/no question.
 *
 * @param _theme  - Active CLI theme (reserved for future accent styling).
 * @param message - Question to display.
 * @param opts    - Optional config; `initial` sets the default selection.
 * @returns       `true` if accepted, `false` if denied.
 * @throws        Error('cancelled') if user presses Ctrl-C.
 *
 * @example
 * ```ts
 * import { buildTheme } from '@hop-top/kit/cli';
 * import { confirm } from '@hop-top/kit/tui';
 * const ok = await confirm(buildTheme(), 'Deploy now?');
 * ```
 */
export async function confirm(
  _theme: Theme,
  message: string,
  opts?: ConfirmOpts,
): Promise<boolean> {
  const result = await clackConfirm({
    message,
    initialValue: opts?.initial,
  });

  if (isCancel(result)) {
    throw new Error('cancelled');
  }

  return result;
}
